plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlScannerTest.java

```diff
@@ -17,20 +17,68 @@
 package com.google.cloud.tools.eclipse.appengine.validation;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+
+import org.eclipse.core.resources.IContainer;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jst.common.project.facet.core.JavaFacet;
+import org.eclipse.jst.j2ee.web.project.facet.WebFacetUtils;
 import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
 import org.junit.Test;
 import org.xml.sax.SAXException;
 import org.xml.sax.ext.Locator2Impl;
 import org.xml.sax.helpers.AttributesImpl;
 
+import com.google.cloud.tools.eclipse.test.util.project.TestProjectCreator;
+import com.google.common.collect.Lists;
+
 public class WebXmlScannerTest {
 
-  private WebXmlScanner scanner = new WebXmlScanner();
+  private static IFile resource;
+  private static IJavaProject javaProject;
+  private WebXmlScanner scanner;
+  
+  @ClassRule public static TestProjectCreator projectCreator = new TestProjectCreator()
+      .withFacetVersions(Lists.newArrayList(JavaFacet.VERSION_1_7, WebFacetUtils.WEB_25));
+  
+  @BeforeClass
+  public static void setUpBeforeClass() throws CoreException {
+    // Project's default source folder is the main project folder.
+    IProject project = projectCreator.getProject();
+    javaProject = projectCreator.getJavaProject();
+    
+    ValidationTestUtils.createFolders(project, new Path("src/main/webapp/WEB-INF"));
+    resource = project.getFile("src/main/webapp/WEB-INF/web.xml");
+    resource.create(new ByteArrayInputStream(new byte[0]), true, null);
+    
+    ValidationTestUtils.createFolders(project, new Path("src/main/java"));
+    IFile servletClass = project.getFile("src/main/java/ServletClass.java");
+    servletClass.create(
+    	new ByteArrayInputStream("public class ServletClass {}".getBytes(StandardCharsets.UTF_8)),
+    	true, null);
+    
+    ValidationTestUtils.createFolders(project, new Path("com/example"));
+    IFile servletClassInPackage = project.getFile("com/example/ServletClassInPackage.java");
```

Comment:
I'll clarify. Before #1591, the Java source folder was always the project root. After #1591, if you add a Java facet, the source folder is "src".

#1591 has been marged. @abbiefarr, you need to merge master into this PR and move the source files under "src".